LSF doesn't gracefully handle rejected job

When an LSF job fails to submit, the harness crashes on an out of range error:
```
Path to Build Dir:<build_dir>
WARNING: Ignoring invalid built-in parameter key launch_id
Submitting job from LSF class using batchfilename run_collective_single_code.sh
bsub  -q batch run_collective_single_code.sh  # This failed because max queueable jobs met
Traceback (most recent call last):
  File "/sw/acceptance/olcf-test-harness/harness/bin/test_harness_driver.py", line 236, in auto_generated_scripts
    submit_exit_value = mymachine.submit_batch_script()
  File "/sw/acceptance/olcf-test-harness/harness/machine_types/base_machine.py", line 221, in submit_batch_script
    exit_status = linux_utilities.submit_batch_script(self,new_env)
  File "/sw/acceptance/olcf-test-harness/harness/machine_types/linux_utilities.py", line 454, in submit_batch_script
    submit_exit_value = a_machine.submit_to_scheduler(batch_script)
  File "/sw/acceptance/olcf-test-harness/harness/machine_types/base_machine.py", line 241, in submit_to_scheduler
    submit_exit_value = self.scheduler.submit_job(batchfilename)
  File "/sw/acceptance/olcf-test-harness/harness/machine_types/lsf.py", line 80, in submit_job
    jobid = jobid_pattern.findall(records[0])[0]
IndexError: list index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/sw/acceptance/olcf-test-harness/harness/bin/test_harness_driver.py", line 527, in <module>
    test_harness_driver()
  File "/sw/acceptance/olcf-test-harness/harness/bin/test_harness_driver.py", line 487, in test_harness_driver
    Vargs.separate_build_stdio)
  File "/sw/acceptance/olcf-test-harness/harness/bin/test_harness_driver.py", line 238, in auto_generated_scripts
    jstatus.log_event(status_file.StatusFile.EVENT_SUBMIT_END, submit_exit_value)
  File "/sw/acceptance/olcf-test-harness/harness/libraries/status_file.py", line 387, in log_event
    event_type, event_subtype, str(event_value))
  File "/sw/acceptance/olcf-test-harness/harness/libraries/status_file.py", line 694, in __log_event
    event_time, event_filename)
  File "/sw/acceptance/olcf-test-harness/harness/libraries/status_file.py", line 875, in get_status_info
    'harness function being executed from wrong directory.')
AssertionError: harness function being executed from wrong directory.
In function subtest._start_test we have a critical error.
The command 'test_harness_driver.py -r -l peak_test/<id>' has exited with a failure.
The exit return value is 1
.
Application osu-microbenchmarks future exception:
Command failed: test_harness_driver.py -r -l peak_test/<id>
```

--- a/harness/machine_types/lsf.py
+++ b/harness/machine_types/lsf.py
@@ -69,17 +69,11 @@ class LSF(BaseScheduler):
         records = submit_stdout.readlines()
         submit_stdout.close()
 
-        #print("records = ")
-        #print(records)
-
-        #print("Extracting LSF jobID from LSF class")
-        jobid_pattern = re.compile('\d+')
-        #print("jobid_pattern = ")
-        #print(jobid_pattern)
-        #print("jobid_pattern.findall = ")
-        jobid = jobid_pattern.findall(records[0])[0]
-        self.set_job_id(jobid)
-        print("LSF jobID = ",self.get_job_id())
+        if p.returncode is 0:
+            jobid_pattern = re.compile('\d+')
+            jobid = jobid_pattern.findall(records[0])[0]
+            self.set_job_id(jobid)
+            print("LSF jobID = ",self.get_job_id())
 
         return p.returncode
 

--- a/harness/machine_types/lsf.py
+++ b/harness/machine_types/lsf.py
@@ -74,6 +74,9 @@ class LSF(BaseScheduler):
             jobid = jobid_pattern.findall(records[0])[0]
             self.set_job_id(jobid)
             print("LSF jobID = ",self.get_job_id())
+        else:
+            with open(temp_stderr,"r") as submit_stderr:
+                print(submit_stderr.read())
 
         return p.returncode
 

--- a/harness/machine_types/pbs.py
+++ b/harness/machine_types/pbs.py
@@ -67,6 +67,9 @@ class PBS(BaseScheduler):
             jobid = jobid_pattern.findall(records[0])[0]
             self.set_job_id(jobid)
             print("PBS jobID = ",self.get_job_id())
+        else:
+            with open(temp_stderr,"r") as submit_stderr:
+                print(submit_stderr.read())
 
         return p.returncode
 

--- a/harness/machine_types/pbs.py
+++ b/harness/machine_types/pbs.py
@@ -62,17 +62,11 @@ class PBS(BaseScheduler):
         records = submit_stdout.readlines()
         submit_stdout.close()
 
-        #print("records = ")
-        #print(records)
-
-        #print("Extracting LSF jobID from PBS class")
-        jobid_pattern = re.compile('\d+')
-        #print("jobid_pattern = ")
-        #print(jobid_pattern)
-        #print("jobid_pattern.findall = ")
-        jobid = jobid_pattern.findall(records[0])[0]
-        self.set_job_id(jobid)
-        print("PBS jobID = ",self.get_job_id())
+        if p.returncode is 0:
+            jobid_pattern = re.compile('\d+')
+            jobid = jobid_pattern.findall(records[0])[0]
+            self.set_job_id(jobid)
+            print("PBS jobID = ",self.get_job_id())
 
         return p.returncode
 

--- a/harness/machine_types/slurm.py
+++ b/harness/machine_types/slurm.py
@@ -69,6 +69,9 @@ class SLURM(BaseScheduler):
             jobid = jobid_pattern.findall(records[0])[0]
             self.set_job_id(jobid)
             print("SLURM jobID = ",self.get_job_id())
+        else:
+            with open(temp_stderr,"r") as submit_stderr:
+                print(submit_stderr.read())
 
         return p.returncode
 

